### PR TITLE
Require Jenkins 2.479.1 or newer

### PR DESCRIPTION
**Pull request by darinpope**
Used OpenRewrite to convert to Java 17 syntax

### Testing done

ran `mvn clean verify`

Confirmed that [coverage badges extension plugin](https://github.com/jenkinsci/coverage-badges-extension-plugin) (the only plugin dependent on this plugin) passes its automated tests with the incremental build of this plugin

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by gounthar — APPROVED** [review]

**Comment by MarkEWaite** [comment]
Thanks!  A similar pull request includes additional changes for the Java EE 8 to Jakarta EE 9 transition.  That pull request is:

* #345 

We'll want to highlight the Java EE  8 to Jakarta EE 9 transition in one of our sessions, since that transition is a key component of the 2.479 release

**Review by MarkEWaite — APPROVED** [review]
Not yet ready to merge and release because we need to wait for more users to adopt 2.479.x, but the code changes look good to me.

File: pom.xml [hunk 1]
```diff
@@ -44,8 +44,8 @@
     <changelist>999999-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
     <!-- https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/ -->
-    <jenkins.baseline>2.452</jenkins.baseline>
-    <jenkins.version>${jenkins.baseline}.4</jenkins.version>
+    <jenkins.baseline>2.479</jenkins.baseline>
```
Comment (gounthar) [inline pom.xml:47]:
```suggestion
    <jenkins.baseline>2.479.1</jenkins.baseline>
```

**Review by gounthar — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (gounthar) [reply pom.xml:47]: Forget it, I hadn't seen the line under this one. 🤷

**Comment by MarkEWaite** [comment]
I think that we've waited long enough.  Merging and releasing.

**Pull request merged**